src/associatedLegendre.jl

```diff
@@ -0,0 +1,594 @@
+println("time SSLegendreP2(100,50,big(.6))  $TP2")
+#time SLegendreP2(100,50,big(.6))  0.0085234
+#time SSLegendreP2(100,50,big(.6))  0.004890456
+# the series is double speed when each term is a multiple of previous term
+
+
+
+"""
+    SSLegendreP3(n::Integer,m::Integer,x::Number)
+    associated Legendre function 1st kind type 3 , x >1.
+    each term in sum is multiple of previous. 
+
+"""
+
+function SSLegendreP3(n::Integer,m::Integer,x::Number)#n,m integers 
+    
+   
+    M = (2*n -1) # M must be odd
+    # (2m-1)!!= (2m)!/( m! 2^m)   double factorial 
+    dblfac=1
+    for j=1:M
+        if  iseven(j)
+            continue
+        end    
+        dblfac=j*dblfac  
+    end
+    
+    MXP=Int(trunc(((n+m)/2)))
+    sum= (x^(n+m))*dblfac/(((x*x - one(x))^(m/2))*prod(1:n-m))
+    prev=sum
+    for p=1:MXP
+        term=(-prev)*(n-p+1)*(n+m-2*p+1)*(n+m-2*p+2
+        )/(p*(2*n-2*p+1)*(2*n-2*p+2)*x*x)
+        sum=sum+term
+        prev=term
+    end
+    return sum
+end
+
+
+   
+
+#checking SSLegendreP3 against Float64(SLegendreP3)
+#   for n< 17 the relative error is less than 4e-14
+# the error depends on the n amd m values and x values to some extent
+"""
+    function testSSLegendreP3F()
+    compare SSLegendreP3 with Float64 SLegendreP3
+"""
+function testSSLegendreP3F()
+    x=2.
+    rerr = 4e-14
+    for n=0:17
+        for m=rand(0:n)
+            a=SSLegendreP3(n,m,x)
+            b=Float64(SLegendreP3(n,m,big(x)))
+            df=norm(a-b)/norm(b) 
+            if ( df > rerr)
+            println(n," ", m," ", a," ",b," ",df)
+            end
+        end
+end
+end
+testSSLegendreP3F()
+
+# based on code published at
+# i. a. selezneva, yu. l. ratis, e. hernÃ¡ndez, j. pÃ©rez-quiles 
+#and p. fernÃ¡ndez de cÃ³rdoba:
+#a code to calculate high order legendre polynomials
+#rev. acad. colomb. cienc.: volumen xxxvii, nÃºmero 145 - diciembre 2013
+#www.scielo.org.co/pdf/racefn/v37n145/v37n145a09.pdf   
+    #Selezneva.....pdf    
+       
+"""
+ZZLegendreP2(n::Integer,m::Integer,z::Number)
+    associated Legendre function First kind type 2 |x|<=1.
+    uses different three term recursion
+"""
+
+function ZZLegendreP2(n::Integer,m::Integer,z::Number)    
+    nz=1
+    pnm = zeros(typeof(z),2*n+1)  
+    fac = prod(2.:n)
+    sqz2 = sqrt((one(z)-z.*z))
+    hsqz2 = 0.5*sqz2
+    ihsqz2 = z./hsqz2
+    if(n==0)
+        pnm[1]=one(z)
+        return ((-one(z))^m)*pnm[n+1+m] #pnm
+    end
+    if(n==1)
+        pnm[1]=-.5*sqz2
+        pnm[2]=z 
+        pnm[3]=sqz2
+        return ((-one(z))^m)*pnm[n+1+m]  #pnm
+    end
+    pnm[1] = (1-2*abs(n-2*floor(n/2)))*hsqz2.^n/fac
+    pnm[2] = -pnm[1]*n.*ihsqz2
+    for mr=1:2*n-1
+        pnm[mr+2]=(mr-n).*ihsqz2.*pnm[mr+1]-(2*n-mr+1)*mr*pnm[mr]
+    end
+    return ((-one(z))^m)*pnm[n+1+m]
+end
+ 
+
+ 
+
+#Benchmarking ZZLegendreP2 against scipy.special
+
+using PyCall
+
+@pyimport scipy.special as s
+
+
+function timePY()
+    for i=1:100
+        s.lpmv(15,30,.6) 
```

Comment:
This benchmark is meaningless; when timing Python functions, you need to time computation whole vector at once.  To minimize the overhead, you would want to do something like (using the BenchmarkTools package):
```
lpmv = pyimport("scipy.special")["lpmv"]
x = PyObject(collect(linspace(0.1,0.9,10^6))) # 10^6 points
m = PyObject(15)
v = PyObject(30)
@btime pycall($lpmv, PyObject, $x, $m, $v)
```